frontend/src/components/ViewTrips/trips-container.js

```diff
@@ -86,19 +43,39 @@ class TripsContainer extends React.Component {
     this.state = {trips: []};
   }
 
-  /** @inheritdoc */
+  /**
+   * When the TripsContainer mounts, a listener is attached to the QuerySnapshot
+   * event that grabs all trip documents where the current user uid is contained
+   * in the collaborator uid array (collaborators field). This allows real-time
+   * updates for all collaborators on a trip whenever a trip is updated (add,
+   * edit, or delete).
+   *
+   * In the case where there is an error, an error component is returned in
+   * place of the array of trips.
+   *
+   * @override
+   */
   async componentDidMount() {
-    try {
-      const querySnapshot = await queryUserTrips(db);
-      let tripsContainer = await serveTrips(querySnapshot,
-                                            this.props.handleEditTrip,
-                                            this.props.refreshTripsContainer);
-      this.setState({ trips: tripsContainer });
-    }
-    catch (error) {
-      let errorElement = await getErrorElement(error);
-      this.setState({ trips: errorElement });
-    }
+    const curUserUid = getCurUserUid();
+    db.collection(DB.COLLECTION_TRIPS)
+        .where(DB.TRIPS_COLLABORATORS, 'array-contains', curUserUid)
+        .orderBy(DB.TRIPS_CREATION_TIME, 'desc')
+        .onSnapshot(querySnapshot => {
+          const tripsContainer = querySnapshot.docs.map(doc =>
+              ( <Trip
+                  tripData={doc.data()}
+                  tripId={doc.id}
+                  handleEditTrip={this.props.handleEditTrip}
+                  key={doc.id}
+                />
+              )
+          );
+
+          this.setState({ trips: tripsContainer });
+        }, (error) => {
+          let errorElement = getErrorElement(error);
+          this.setState({ trips: errorElement });
+        });
```

Comment:
Is this all that's required for the realtime stuff? whoa.

Reply:
Yeah I was surprised at how little difference there is.